src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -964,13 +952,263 @@ def load_module(name, value):
+                config_is_cached = True
+
+            # retrieve all folder_names that contain relevant files
+            folder_names = [k for k, v in config_dict.items() if isinstance(v, list)]
+
+            filenames = set(sibling.rfilename for sibling in info.siblings)
+            model_filenames, variant_filenames = variant_compatible_siblings(filenames, variant=variant)
+
+            # if the whole pipeline is cached we don't have to ping the Hub
+            if revision in DEPRECATED_REVISION_ARGS and version.parse(
+                version.parse(__version__).base_version
+            ) >= version.parse("0.15.0"):
+                warn_deprecated_model_variant(
+                    pretrained_model_name_or_path, use_auth_token, variant, revision, model_filenames
+                )
+
+            model_folder_names = set([os.path.split(f)[0] for f in model_filenames])
+
+            # all filenames compatible with variant will be added
+            allow_patterns = list(model_filenames)
+
+            # allow all patterns from non-model folders
+            # this enables downloading schedulers, tokenizers, ...
+            allow_patterns += [os.path.join(k, "*") for k in folder_names if k not in model_folder_names]
+            # also allow downloading config.jsons with the model
+            allow_patterns += [os.path.join(k, "*.json") for k in model_folder_names]
+
+            allow_patterns += [
+                SCHEDULER_CONFIG_NAME,
+                CONFIG_NAME,
+                cls.config_name,
+                CUSTOM_PIPELINE_FILE_NAME,
+            ]
+
+            if from_flax:
+                ignore_patterns = ["*.bin", "*.safetensors", "*.onnx", "*.pb"]
+            elif is_safetensors_available() and is_safetensors_compatible(model_filenames, variant=variant):
+                ignore_patterns = ["*.bin", "*.msgpack"]
+
+                safetensors_variant_filenames = set([f for f in variant_filenames if f.endswith(".safetensors")])
+                safetensors_model_filenames = set([f for f in model_filenames if f.endswith(".safetensors")])
+                if (
+                    len(safetensors_variant_filenames) > 0
+                    and safetensors_model_filenames != safetensors_variant_filenames
+                ):
+                    logger.warn(
+                        f"\nA mixture of {variant} and non-{variant} filenames will be loaded.\nLoaded {variant} filenames:\n[{', '.join(safetensors_variant_filenames)}]\nLoaded non-{variant} filenames:\n[{', '.join(safetensors_model_filenames - safetensors_variant_filenames)}\nIf this behavior is not expected, please check your folder structure."
+                    )
+            else:
+                ignore_patterns = ["*.safetensors", "*.msgpack"]
+
+                bin_variant_filenames = set([f for f in variant_filenames if f.endswith(".bin")])
+                bin_model_filenames = set([f for f in model_filenames if f.endswith(".bin")])
```

Comment:
can directly use a set-comprehension here:

```suggestion
                bin_variant_filenames = {f for f in variant_filenames if f.endswith(".bin")}
                bin_model_filenames = {f for f in model_filenames if f.endswith(".bin")}
```